docs/registering-strategies.adoc

```diff
@@ -0,0 +1,87 @@
+== Registering Custom Strategies
+
+By default *Smart Testing* comes with a set of strategies such as `new`, `changed`, `affected` or `failed`, but you can implement your own strategies and registering them.
+
+To do it you just need to follow next four steps:
+
+* Implement `org.arquillian.smart.testing.spi.TestExecutionPlannerFactory` Java SPI and registering as such.
+* Implement `org.arquillian.smart.testing.spi.TestExecutionPlanner` interface.
+* Create a `.properties` file with name strategy as _key_ part, and maven coordinates of strategy as _value_ part. (optional)
+* Configures `.properties` location in *Smart Testing*. (optional)
+
+=== Implementing `TestExecutionPlannerFactory`
+
+`TestExecutionPlannerFactory` is a Java SPI interface that manages the creation of the strategy.
+You need to implement three simple methods, two related with the name of the strategy and one that generates the strategy class.
+
+As an example from `failed` strategy:
+
+[source, java]
+.FailedTestsDetectorFactory.java
+----
+include::../strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/FailedTestsDetectorFactory.java[tag=documentation]
+----
+<1> Returns the alias of the strategy
+<2> Returns an instance of `TestExecutionPlanner`
+
+Of course as any other Java SPI, you need to register it in `META-INF/services`.
+
+[source, properties]
+.META-INF/services/org.arquillian.smart.testing.spi.TestExecutionPlannerFactory
+----
+include::../strategies/failed/src/main/resources/META-INF/services/org.arquillian.smart.testing.spi.TestExecutionPlannerFactory[]
+----
+
+=== Implementing `TestExecutionPlanner`
+
+`TestExecutionPlanner` is the strategy implementation of the strategy.
+The important method there is `getTests()` which returns a collection of the tests that must be executed.
+You also need to set there the strategy name.
+
+For example, failed strategy is implemented as:
+
+[source, java]
+.FailedTestsDetector.java
+----
+include::../strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/FailedTestsDetector.java[tag=documentation]
+----
+<1> Method that returns list of tests to execute
+<2> `TestSelection` requires fully qualified test class name and the strategy where is calculated
+
+=== Create `.properties` file
+
+Finally you need to create a `.properties` file where you set the strategy name as key and the Maven coordinates as value.
+This file is required so *Smart Testing* can automatically register the artifacts in build tool script.
+
+As an example of registering previous strategy:
+
+[[custom-strategy]]
+[source, properties]
+.custom-smart-testing.properties
+----
+smart.testing.strategy.failed=org.arquillian.smart.testing:strategy-failed:1.0.0
```

Comment:
as a first example I would use some non-existing strategy eg:
`smart.testing.strategy.cool=my.company.smart.testing:strategy-cool:1.0.0`
then I would say that this strategy can be activated using property value `cool`
Then, you could also show how it is done for our existing strategies...
This is just my opinion ;-)

Reply:
Yes the thing is that if I pick an existing strategy then I can include directly the code, so if it changes the interface names, then the documentation is automatically updated.